#!/usr/bin/env python
PACKAGE = "stl_frontier_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#           Name                        Type       Level  Description                                                     Default       Min     Max
gen.add("lambda",                       double_t,       1,   "Lambda for LTL",                                            5.0,          0.0,    10.0)
gen.add("min_distance",                 double_t,       2,   "Min distance for LTL",                                      1.0,          0,      20.0)
gen.add("max_distance",                 double_t,       2,   "Max distance for LTL",                                      4.0,          0,      20.0)
gen.add("min_distance_active",          bool_t,         3,   "True if min distance for LTL is active",                    True)
gen.add("max_distance_active",          bool_t,         3,   "True if max distance for LTL is active",                    True)
gen.add("routers_active",               bool_t,         3,   "True if routers for LTL is active",                         True)
gen.add("distance_add_path",            double_t,       4,   "Distance travelled to add to path",                         0.5,          0,      10.0)
gen.add("max_search_distance",          double_t,       5,   "Maximum distance to search for obstacles",                  20.0,         0,      100.0)
gen.add("step_size",                    double_t,       6,   "Step size",                                                 0.1,          0,      1.0)

exit(gen.generate(PACKAGE, "stl_frontier_planner", "STL"))
